### PR TITLE
Added subtractFee to SendTransaction RPC-method

### PR DESCRIPTION
**Pull request by schulterklopfer**
added subtractFee as a send rpc command param, so we can configure if mining fee is included or on top of the amount to send

### CODE REVIEW

**Comment by lontivero** [comment]
This cannot work because the fee can be subtracted from only one output but this PR tries to subtract it from all the outputs.
Could you update the RPC.md document too?

**Comment by schulterklopfer** [comment]
Oh. I see the select now and we tested it yesterday with only 1 element in the payments array.
(brainfart :-D)

**Comment by schulterklopfer** [comment]
Added the documentation change. Changed the code so it takes the fee from the highest valued output (uncommitted, needs more testing)

**Comment by lontivero** [comment]
> Changed the code so it takes the fee from the highest valued output.

What if someone else want to subtract the fee from a different output? I can only accept changes that can eventually be accepted and merged to master. If you need to be able to subtract from output then do something like this:

```bash
curl -s http://127.0.0.1:18099/ --data-binary @- << EOF
{
    "jsonrpc":"2.0",
    "id":"1",
    "method":"send", 
    "params": { 
        "payments":[ {
            "sendto": "tb1qg32kp05hs8mny954hp2snvnht40a08guxczmwh", 
            "amount": 50000, 
            "label": "Marcela"
       }, {
           "sendto":"tb1qmmnc35p3dppyhfrpys6skr27zlv9ew47qmk0gn", 
           "amount": 900000, 
           "label": "Joaquin",
           "subtractFee": true
       } ], 
       "coins":[{}],
      "feeTarget":2
    }
} 
EOF 

```

**Comment by schulterklopfer** [comment]
Will do! :D ..
Thanks for your input!

File: WalletWasabi.Gui/Rpc/WasabiJsonRpcService.cs [hunk 1]
```diff
@@ -98,14 +98,15 @@ public class Payment
...
+			public bool subtractFee;
 		}
 
 		[JsonRpcMethod("send")]
 		public async Task<object> SendTransaction(Payment[] payments, TxoRef[] coins, int feeTarget)
 		{
 			AssertWalletIsLoaded();
 			var sync = _global.Synchronizer;
-			var payment = new PaymentIntent( payments.Select(p=> new DestinationRequest(p.sendto.ScriptPubKey, MoneyRequest.Create(p.amount), new SmartLabel(p.label))));
+			var payment = new PaymentIntent(payments.Select((p,i)=> new DestinationRequest(p.sendto.ScriptPubKey, MoneyRequest.Create(p.amount, p.subtractFee), new SmartLabel(p.label))));
```
Comment (lontivero) [inline WalletWasabi.Gui/Rpc/WasabiJsonRpcService.cs]: Remove the `i` and I will merge it.

**Review by lontivero — CHANGES_REQUESTED** [review]

**Pull request merged**